src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/utils.ts

```diff
@@ -0,0 +1,65 @@
+import { TFunction } from 'react-i18next';
+
+import { produceVMDisks } from '@catalog/utils/WizardVMContext';
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import {
+  V1beta1DataVolumeSpec,
+  V1ContainerDiskSource,
+  V1VirtualMachine,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { CDI_BIND_REQUESTED_ANNOTATION } from '@kubevirt-utils/hooks/useCDIUpload/consts';
+import { getTemplateContainerDisks } from '@kubevirt-utils/resources/template';
+import { getDisks, getVolumes } from '@kubevirt-utils/resources/vm';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+
+export const getRegistryHelperText = (template: V1Template, t: TFunction) => {
+  const containerDisks = getTemplateContainerDisks(template);
+
+  if (containerDisks && containerDisks.length > 0)
+    return t('Enter a container image (for example: {{containerDisk}})', {
+      containerDisk: containerDisks[0],
+    });
+};
+
+export const getDiskSource = (
+  vm: V1VirtualMachine,
+  diskName: string,
+): undefined | V1beta1DataVolumeSpec | V1ContainerDiskSource => {
+  if (!diskName) return;
+
+  const disk = getDisks(vm)?.find((d) => d.name === diskName);
+  const volume = getVolumes(vm)?.find((v) => v.name === disk?.name);
+
+  if (!disk || !volume) return;
+
+  if (volume.containerDisk) {
+    return volume.containerDisk;
+  }
+
+  if (volume.dataVolume) {
+    const dataVolumeTemplate = vm.spec?.dataVolumeTemplates?.find(
+      (template) => template.metadata?.name === volume.dataVolume.name,
+    );
+
+    return dataVolumeTemplate?.spec;
+  }
+};
+
+export const overrideVirtualMachineDataVolumeSpec = (
+  virtualMachine: V1VirtualMachine,
+  customSource?: V1beta1DataVolumeSpec,
+): V1VirtualMachine => {
+  return produceVMDisks(virtualMachine, (draftVM) => {
+    const dataVolumeTemplate = draftVM.spec.dataVolumeTemplates[0];
```

Comment:
is it pormised that there will result at [0]?

Reply:
Okay, if there is no dataVolumeTemplate, I'll create one here